### PR TITLE
POI Spawning: Handle empty preset list, restrict protos in NfAdventureRule

### PR DESCRIPTION
**Pull request by whatston3**
## About the PR

* Better handles empty preset lists in NfAdventureRuleSystem and PointOfInterestSystem
* ~~Removes the preset lists from most POIs (note: not the cove).~~ Reverted this.
  * ~~If this should be returned to how it was ("we don't want to support secret/deathmatch/X"), easily done, no harm no foul.~~ Reverted this.  No harm, no foul.
* Adds a comment to the PointOfInterestPrototype that empty preset lists are treated as accepting any preset
* Reorganized namespace/system dependencies in alphabetical order
  * ~~milon~~ Reverted this.

## Why / Balance

Empty preset lists should be supported in-spec, and it makes sense (at least to me) to have it as working in all presets if the NfAdventureRule is present.

If this was the case, #2613 wouldn't have been needed.

## How to test

Run the game, check that all of the POI stations are present, and the cove is present if running NFPirates.

## Media

Behold, stations.
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

No player-facing changes, no changelog.

### CODE REVIEW

File: Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs [hunk 1]
```diff
@@ -28,14 +29,17 @@ namespace Content.Server._NF.GameRule;
 /// </summary>
 public sealed class NFAdventureRuleSystem : GameRuleSystem<NFAdventureRuleComponent>
 {
-    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
     [Dependency] private readonly IConfigurationManager _configurationManager = default!;
     [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
```
Comment (MilonPL) [inline Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs:34]: `_cfg`, `_player` and `_proto`

File: Content.Server/_NF/GameRule/PointOfInterestSystem.cs [hunk 2]
```diff
@@ -23,11 +23,11 @@ public sealed class PointOfInterestSystem : EntitySystem
     [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
     [Dependency] private readonly IRobustRandom _random = default!;
     [Dependency] private readonly IConfigurationManager _configurationManager = default!;
```
Comment (MilonPL) [inline Content.Server/_NF/GameRule/PointOfInterestSystem.cs:25]: I'm pretty sure C comes before P
also shorten the names

**Review by MilonPL — COMMENTED** [review]
the curse

File: Content.Server/_NF/GameRule/PointOfInterestSystem.cs (hunk 2 above)
Comment (whatston3) [reply Content.Server/_NF/GameRule/PointOfInterestSystem.cs]: unexpected kurwaball

**Review by dvir001 — APPROVED** [review]

**Pull request merged**